### PR TITLE
3988 get_project_status_for_user should return only visible tasks

### PR DESCRIPTION
**Pull request by sophiemachin**
Update project_list to contain only projects that both
- are visible
- have tasks

Update the project task list to only contain tasks that are visible. 

Accept a `include_non_visible_data` switch will show all tasks and projects regardless of visibility status

Update test_project_status_for_user to remove any visible projects from project lists.

Add a test for the `include_non_visible_data` flag

### CODE REVIEW

File: api/endpoints/project.py [hunk 1]
```diff
@@ -252,27 +252,38 @@ def calculate_task_url(self, task, task_id):
 
     def main(self):
         try:
+            projects = []
             for project in self.project_list:
                 project["project_is_visible"] = self.calculate_project_visibility(
                     project
                 )
+
```
Comment (afs25) [inline api/endpoints/project.py:260]: Performance can be slightly improved by moving the conditional statement `if project["project_is_visible"]` from line 265 to here.

**Review by afs25 — COMMENTED** [review]
Other than the performance tuning I noted inline, Andy left a comment on this user story asking for a include_non_visible_data flag that would make the API return everything (revert to previous behaviour). Did you give some thought to that and decide not to implement? We can discuss at the next dev meeting if helpful.

**Comment by sophiemachin** [comment]
> Other than the performance tuning I noted inline, Andy left a comment on this user story asking for a include_non_visible_data flag that would make the API return everything (revert to previous behaviour). Did you give some thought to that and decide not to implement? We can discuss at the next dev meeting if helpful.

Ah yes I wasn't sure of the protocol for scope creep - my old job we always finished the work and then did another PR. I can add it to this if you prefer?

File: api/endpoints/project.py (hunk 1 above)
Comment (sophiemachin) [reply api/endpoints/project.py]: Fixed

**Comment by afs25** [comment]
> Ah yes I wasn't sure of the protocol for scope creep - my old job we always finished the work and then did another PR. I can add it to this if you prefer?

Happy for you to address this in whatever way you are most comfortable. Another PR is absolutely fine (I do that too sometimes). I was just confirming you saw that comment.

**Comment by sophiemachin** [comment]
I've put it in with this PR because it's not a huge change. I've added a new test to check the switch works. The deployed code also works with Postman.

**Review by afs25 — APPROVED** [review]

**Pull request merged**